train_segmentation_model.py

```diff
@@ -191,7 +202,9 @@ def train(gcp_bucket, config_file, random_module_global_seed, numpy_random_globa
         'global_threshold_for_metrics': global_threshold,
         'random-module-global-seed': random_module_global_seed,
         'numpy_random_global_seed': numpy_random_global_seed,
-        'tf_random_global_seed': tf_random_global_seed
+        'tf_random_global_seed': tf_random_global_seed,
+        'pre_trained_model_id': pre_trained_model_id,
+        'pre_trained_model_config': pre_trained_model_config
```

Comment:
we may also want pre trained model metadata, given that it'll contain additional details like seeds. also this should specify train config vs. dataset config

Reply:
Can you please clarify? the metadata from the pretrained model is already included in the metadata of the new model

Reply:
i see the pretrained config (pretrain input) but not the pretrained metadata (pretrain output)

Reply:
like this? 
![image](https://user-images.githubusercontent.com/43469359/97042395-f8b99680-153e-11eb-98e3-08e71c0ae4bc.png)

Reply:
yup that's awesome. just convert `pre_train` to `pretrain` then. 

also, for readability of the new model metadata, i would suggest we group all pretrained stuff under one dict key in the current model metadata dictionary (kind of like in train seg thresholds), which requires just a quick rewrite to form in advance the pretrained model id, config, and metadata into a separate dictionary first, defined like

```
 pretrained_model_info={'pretrained_id': pretrained_id, 'pretrained_config': pretrained_config, 'pretrained_meta': pretrained_meta}
```

and then current model metadata is just 

```
metadata = {
    the usual stuff...,
    'pretrained_model_info' : pretrained_model_info
}
```
or however u want to label things. since with current way, unordered current metadata dict could shuffle the pretrained content

Reply:
![image](https://user-images.githubusercontent.com/43469359/97046601-9fa13100-1545-11eb-81f7-513f5324d27d.png)
 done!

Reply:
looks great, last thing, didn't mean to suggest the labelling exactly, so u can keep `pretrained_model_metadata` or just `pretrained_metadata` (vs. `meta`), same for config and id. besides this, solid! u can imagine how this will nest nicely now if we pretrain a model an arbitrary number of times